qcodes/dataset/experiment_settings.py

```diff
@@ -0,0 +1,60 @@
+"""Settings that are indirectly related to experiments."""
+
+from typing import Optional
+from qcodes.dataset.sqlite.connection import ConnectionPlus, atomic_transaction
+
+# The default experiment's exp_id. This changes to the exp_id of a created/
+# loaded experiment.
+default_experiment: Optional[int] = None
+
+
+def _set_default_experiment_id(exp_id: int) -> None:
+    """
+    Sets the default_experiment to the exp_id of a created/ loaded experiment.
+
+    Args:
+        exp_id: The exp_id of an experiment.
+    """
+    global default_experiment
+    default_experiment = exp_id
+
+
+def _get_latest_default_experiment_id() -> Optional[int]:
+    """
+    Gets the lastest created or loaded experiment's exp_id.
+
+    Returns:
+        The latest created/ loaded experiment's exp_id.
+    """
+    global default_experiment
+    return default_experiment
+
+
+def reset_default_experiment_id() -> None:
+    """
+    Resets the default_experiment to None.
+    """
+    global default_experiment
+    default_experiment = None
+
+
+def get_default_experiment_id(conn: ConnectionPlus) -> Optional[int]:
+    """
+    Returns the latest created/ loaded experiment's exp_id as the default
+    experiment. If it is None, maximum exp_id from the currently active
+    initialized database is returned as the default. if no experiment
+    found in the database, a ValueError is raised.
+
+    Returns:
+        exp_id of the default experiment.
+    """
+    exp_id = _get_latest_default_experiment_id()
+    if exp_id is None:
+        query = "SELECT MAX(exp_id) FROM experiments"
+        c = atomic_transaction(conn, query)
+        exp_id = c.fetchall()[0][0]
+    if exp_id is None:
+        raise ValueError("No experiments found."
+                         " You can create one with:"
+                         " new_experiment(name, sample_name)")
```

Comment:
This code is duplicated from the queries module where it belongs. Importing that directly will create a circular import since that module imports connect from the database module (same module that defines the initialise_database function which uses a function from this module) The only reason for the import of connect is the function `get_guids_from_multiple_run_ids` which is implemented differently from all other queries. IMHO we should just break this function and put it inline with all the other ones by taking a connection and not a dbpath as input.

Reply:
Good catch. We can break the dependency and import get_last_experiment directly

Reply:
This still need to make use of the query. Please do not resolve until this is done